packages/@aws-cdk/aws-gamelift/lib/build.ts

```diff
@@ -0,0 +1,191 @@
+
+  /**
+     * Version of this build
+     *
+     * @default No version
+     */
+  readonly buildVersion?: string;
+
+  /**
+     * The operating system that the game server binaries are built to run on.
+     *
+     * @default No version
+     */
+  readonly operatingSystem?: OperatingSystem;
+
+  /**
+     * The game build file storage
+     */
+  readonly content: Content;
+
+  /**
+    * The IAM role assumed by GameLift to access server build in S3.
+    * If providing a custom role, it needs to trust the GameLift service principal (gamelift.amazonaws.com) and be granted sufficient permissions.
+    *
+    * @see https://docs.aws.amazon.com/gamelift/latest/developerguide/security_iam_id-based-policy-examples.html#security_iam_id-based-policy-examples-access-storage-loc
+    *
+    * @default - a role will be created with default permissions.
+    */
+  readonly role?: iam.IRole;
+}
+
+/**
+ * Create a GameLift server build
+ *
+ * @resource AWS::GameLift::Build
+ */
+export class Build extends BuildBase {
+
+  /**
+   * Create a new Build from s3 content
+   */
+  static fromBucket(scope: Construct, id: string, bucket: s3.IBucket, key: string, objectVersion?: string) {
+    return new Build(scope, id, {
+      content: Content.fromBucket(bucket, key, objectVersion),
+    });
+  }
+
+  /**
+   * Create a new Build from asset content
+   */
+  static fromAsset(scope: Construct, id: string, path: string, options?: s3_assets.AssetOptions) {
+    return new Build(scope, id, {
+      content: Content.fromAsset(path, options),
+    });
+  }
+
+  /**
+     * Import a build into CDK using its identifier
+     */
+  static fromBuildId(scope: Construct, id: string, buildId: string): IBuild {
+    return this.fromBuildAttributes(scope, id, { buildId });
+  }
+
+  /**
+   * Import an existing build from its attributes.
+   */
+  static fromBuildAttributes(scope: Construct, id: string, attrs: BuildAttributes): IBuild {
+    class Import extends BuildBase {
+      public readonly buildId = attrs.buildId;
+      public readonly grantPrincipal = attrs.role ?? new iam.UnknownPrincipal({ resource: this });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The Identifier of the build.
+   */
+  public readonly buildId: string;
+
+  /**
+   * The IAM role GameLift assumes to acccess server build content.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * The principal this GameLift Build is using.
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  constructor(scope: Construct, id: string, props: BuildProps) {
+    super(scope, id, {
+      physicalName: props.buildName,
+    });
+
+    if (props.buildName) {
+      if (props.buildName.length > 1024) {
+        throw new Error(`Build name can not be longer than 1024 characters but has ${props.buildName.length} characters.`);
+      }
+    }
+    this.role = props.role ?? new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('gamelift.amazonaws.com'),
```

Comment:
I am granting read access to for an S3 bucket + dedicated content key on binding phase for a content (see content.ts on bind methods) but otherwise no specific needs here or any specific policy